### DISCUSSION TITLE
Can't generate multiple services imported in the main.tsp

### DISCUSSION THREAD

**Discussion by Zakharii-Husar** [microsoft/typespec#4224] [OpenAPI3]
Have been stuck for a while on this problem: every time I'm trying to emit a yaml file by running `npx tsp compile  .`  I get generated a yaml file only for the last imported service. Here's my  tspconfig.yaml :  
``` emit:
  - "@typespec/openapi3"

options:
  "@typespec/openapi3":
    "output-file": "openapi.yaml"
```

I don't know if it has anything to do with it but I also was getting warnings during compilation ln terminal: `No namespace with '@service' was found, but Namespace 'AdminService' contains routes. Did you mean to annotate this with '@service'?
` which I found unreasonable, since I have service decorators everywhere, and added `#suppress "@typespec/http/no-service-found" ""` to supress it. Here's the [repository](https://github.com/Zakharii-Husar/e-commerce/tree/main/typespec) I'm having issues with.

**Comment by bterlson** [comment]
I will try to test out the repo soon, but if you can produce something that repros in the playground that would be a huge help.

In the meantime, one thing I see that is new to me is the nested service namespaces. Are you trying to get standalone OpenAPI specs for each of those? It should work, just curious.

One thing that won't work though is the way you're using `using`. Using is not like an import, it only makes the used symbols available. I think if you want to adjust what you have you would want each namespace to be nested under `ECommerce`, so e.g. `namespace ECommerce.AuthService;`. Then you're done, no need for the using.

**Reply by Zakharii-Husar** [reply]
Thank you for the reply. Honestly it's my first time using TypeSpec.io so I wasn't sure how to separate services  in different files that's why I made a mistake with "using" keyword. Can not see a good way to reproduce this issue in the playground because It doesn't support creation of different files. And the problem for me is with that: all services work and can be compiled into one yaml file but not at the same time. Always output includes only 1 of the imported services. Tried to read documentation on namespaces and emitters but  genuinely don't know what I'm doing wrong.

**Reply by Zakharii-Husar** [reply]
```
import "./services/adminService.tsp";
import "./services/authService.tsp";
import "./services/ordersService.tsp";
import "./services/productsService.tsp";
import "./services/usersService.tsp"; 
import "./services/wishlistService.tsp";
import "./services/cartService.tsp";
import "./services/sellerService.tsp";
```

That's the order I'm importing in. And  it will compile only Seller's service to the yaml file. But If I import for example cartService.tsp as last, then it will compile only the cart service. I wonder why It depends on the order and It always compiles only the last service?

**Comment by Zakharii-Husar** [comment]
Brian thank you for taking a look at my code and confirming that it was supposed to work, just solved the problem today: got a new warning out of the blue in my text editor and found  out that my global TypeSpec compiler conflicted with local version, so reinstalling the template and copy pasting existing code helped to solve it.
